src/EnergyPlus/SwimmingPool.cc

```diff
@@ -105,29 +105,23 @@ namespace EnergyPlus::SwimmingPool {
 // 4. Smith, C., R. Jones, and G. Lof (1993). Energy Requirements and Potential Savings for Heated
 //    Indoor Swimming Pools. ASHRAE Transactions 99(2), p.864-874.
 
-void SimSwimmingPool(EnergyPlusData &state, bool FirstHVACIteration)
+SwimmingPoolData *SwimmingPoolData::factory(EnergyPlusData &state, std::string const &objectName)
 {
-    // Process the input data if it hasn't been done already
     if (state.dataSwimmingPools->getSwimmingPoolInput) {
         GetSwimmingPool(state);
         state.dataSwimmingPools->getSwimmingPoolInput = false;
     }
-
-    // System wide (for all pools) inits
-    state.dataHeatBalFanSys->SumConvPool = 0.0;
-    state.dataHeatBalFanSys->SumLatentPool = 0.0;
-
-    PlantLocation A(0, DataPlant::LoopSideLocation::Invalid, 0, 0);
-    Real64 CurLoad = 0.0;
-    bool RunFlag = true;
-
-    for (auto &thisPool : state.dataSwimmingPools->Pool) {
-        thisPool.simulate(state, A, FirstHVACIteration, CurLoad, RunFlag);
+    // Now look for this particular swimming pool in the list
+    for (auto &pool : state.dataSwimmingPools->Pool) {
+        if (pool.Name == objectName) {
+            return &pool;
+        }
     }
-
-    if (state.dataSwimmingPools->NumSwimmingPools > 0) HeatBalanceSurfaceManager::CalcHeatBalanceInsideSurf(state);
-
-    ReportSwimmingPool(state);
+    // If we didn't find it, fatal
+    ShowFatalError(state,
+                   format("LocalSwimmingPoolFactory: Error getting inputs or index for swimming pool named: {}", objectName)); // LCOV_EXCL_LINE
+    // Shut up the compiler
+    return nullptr; // LCOV_EXCL_LINE
```

Comment:
Yep, looks like a typical plant factory.  I like that it returns a SwimmingPoolData* instead of a PlantComponent*.  That makes things easier.